tests/Doctrine/Tests/DBAL/Functional/ResultCacheTest.php

```diff
@@ -168,6 +169,43 @@ public function testDontFinishNoCache() : void
         self::assertCount(2, $this->sqlLogger->queries);
     }
 
+    /**
+     * @param array<string, int> $params
+     * @param array<string, int> $types
+     *
+     * @dataProvider provideParamsAndTypes
+     */
+    public function testCacheFetchWithParamsAndTypes(array $params, array $types) : void
+    {
+        $sqlQuery = 'SELECT * FROM caching WHERE test_int = :number';
+
+        $stmt = $this->connection->executeQuery($sqlQuery, $params, $types, new QueryCacheProfile(10));
+
+        $stmt->fetch(FetchMode::ASSOCIATIVE);
+        $stmt->closeCursor();
+
+        $stmt = $this->connection->executeQuery($sqlQuery, $params, $types, new QueryCacheProfile(10));
+
+        $stmt->fetch(FetchMode::ASSOCIATIVE);
+        $stmt->closeCursor();
+
+        $stmt = $this->connection->executeQuery($sqlQuery, ['number' => 100], [], new QueryCacheProfile(10));
+
+        $stmt->fetch(FetchMode::ASSOCIATIVE);
+        $stmt->closeCursor();
+
+        self::assertCount(2, $this->sqlLogger->queries);
+    }
+
+    /**
+     * @return iterable<array<string, int>, array<string, int>>
+     */
+    public static function provideParamsAndTypes() : iterable
+    {
+        yield [['number' => 300], []];
+        yield [['number' => 300], ['number' => PDO::PARAM_INT]];
```

Comment:
This test may fail on `3.0.x` due to the usage of `PDO::PARAM_INT`. Does it have to be the deprecated type?

Reply:
No, any type will do here because it is just used to test if it will be used for caching. I'll change it.

Reply:
Does PDO string work?

Reply:
Does it have to be PDO? `ParameterType::STRING` is the recommended analog.

Reply:
`ParameterType::STRING` is now in use.